### PR TITLE
Fix setting multiple cookie headers in the response

### PR DESCRIPTION
**Pull request by sandydoo**
Patches #33 and https://github.com/simplabs/ember-cookies/issues/133.

Cookies were previously set together with the other headers. This meant that if you were to set more than one cookie in FastBoot, each subsequent cookie would overwrite the `Set-Cookie` header and only the last cookie would be sent in the response. This particularly affects `ember-simple-auth`.

This commit extracts the cookie headers, dedups them by name and sets the header at once to set multiple cookies.

This doesn't use `res.cookie` because we're already dealing with serialized headers.
Should be compliant down to node 4.

#### Todo:
* [ ] Set cookies in the test app to make the tests pass. How do we do this?

### CODE REVIEW

**Comment by morgano86** [comment]
Any update on this?

**Comment by stephankaag** [comment]
Ping @rwjblue :-)

**Comment by simonihmig** [comment]
Also waiting for this...

File: src/index.js [hunk 1]
```diff
@@ -37,16 +36,27 @@ function fastbootExpressMiddleware(distPath, options) {
       responseBody.then(body => {
         let headers = result.headers;
         let statusMessage = result.error ? 'NOT OK ' : 'OK ';
-    
-        for (var pair of headers.entries()) {
-          res.set(pair[0], pair[1]);
+        let cookies = {};
+
+        for (let pair of headers.entries()) {
```
Comment (rwjblue) [inline src/index.js]:
```suggestion
        for (let [name, value] of headers.entries()) {
```

File: src/index.js [hunk 2]
```diff
@@ -37,16 +36,27 @@ function fastbootExpressMiddleware(distPath, options) {
...
+          let name = pair[0];
+          let value = pair[1];
+
+          if (pair[0].toLowerCase() === 'set-cookie') {
```
Comment (rwjblue) [inline src/index.js]:
```suggestion
          if (name.toLowerCase() === 'set-cookie') {
```

File: src/index.js [hunk 3]
```diff
@@ -37,16 +36,27 @@ function fastbootExpressMiddleware(distPath, options) {
...
+            let cookieName = value.split('=')[0];
+            cookies[cookieName] = value;
+          } else {
+            res.set(name, value);
+          }
         }
-    
+
+        res.set('Set-Cookie', Object.keys(cookies).map(k => cookies[k]));
```
Comment (rwjblue) [inline src/index.js]: Presumably we only want to do this when cookies are present?

```suggestion
        if (cookies.length > 0) {
          res.set('Set-Cookie', cookies.values());
        }
```

File: src/index.js [hunk 4]
```diff
@@ -37,16 +36,27 @@ function fastbootExpressMiddleware(distPath, options) {
       responseBody.then(body => {
         let headers = result.headers;
         let statusMessage = result.error ? 'NOT OK ' : 'OK ';
-    
-        for (var pair of headers.entries()) {
-          res.set(pair[0], pair[1]);
+        let cookies = {};
```
Comment (rwjblue) [inline src/index.js]:
```suggestion
        let cookies = new Map();
```

File: src/index.js [hunk 5]
```diff
@@ -37,16 +36,27 @@ function fastbootExpressMiddleware(distPath, options) {
...
+
+        for (let pair of headers.entries()) {
+          let name = pair[0];
+          let value = pair[1];
+
+          if (pair[0].toLowerCase() === 'set-cookie') {
+            let cookieName = value.split('=')[0];
+            cookies[cookieName] = value;
```
Comment (rwjblue) [inline src/index.js]:
```suggestion
            cookies.set(cookieName, value);
```

**Comment by sandydoo** [comment]
> Made some suggested changes inline, definitely 👍 on getting this fixed...

🔥 Great stuff! Any thoughts on how we can get a working test for this?

**Comment by sandydoo** [comment]
Closing since #60 has been merged. Thanks, @bobisjan and @rwjblue!

**Pull request closed without merging**